js/src/base-component.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * --------------------------------------------------------------------------
+ * Bootstrap (v5.0.0-alpha3): base-component.js
+ * Licensed under MIT (https://github.com/twbs/bootstrap/blob/main/LICENSE)
+ * --------------------------------------------------------------------------
+ */
+
+import Data from './dom/data'
+
+class BaseComponent {
+  constructor(element) {
+    if (!element) {
+      return
+    }
+
+    this._element = element
+    Data.setData(element, this.constructor.DATA_KEY, this)
+  }
+
+  dispose() {
+    Data.removeData(this._element, this.constructor.DATA_KEY)
+    this._element = null
+  }
+
+  /** Static */
+
+  static getInstance(element) {
+    return Data.getData(element, this.DATA_KEY)
+  }
+
+  static get DATA_KEY() {
```

Comment:
Does this make sense to have it here if we are just returning null and we add `GET_DATA` is all components?

Reply:
I created the BaseComponent as an abstract class that's why I added that to be consistent but that's true currently abstract class doesn't exist in JavaScript

Reply:
Maybe we should remove this part then? The rest look good to me :)

Reply:
done